[vtk] Use a custom hidden namespace for vtkdoubleconversion

This situation is unique to Drake, so we won't to upstream this patch.

--- ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h
+++ ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h
@@ -31,7 +31,7 @@
 // XXX(kitware): // include exports header
 #include "vtkdoubleconversion_export.h"
 // mangle the double_conversion namespace
-#define double_conversion vtkdouble_conversion
+#define double_conversion drake_vendor_double_conversion __attribute__((visibility("hidden")))
 
 #include <cstdlib>
 #include <cstring>

--- Common/Core/vtkNumberToString.cxx
+++ Common/Core/vtkNumberToString.cxx
@@ -7,6 +7,11 @@
 #include VTK_DOUBLECONVERSION_HEADER(double-conversion.h)
 // clang-format on
 
+#undef double_conversion
+namespace double_conversion {
+using namespace drake_vendor_double_conversion;
+}
+
 #include <array>
 #include <sstream>
 
